pkg/slayers/path/scion/raw.go

```diff
@@ -129,36 +129,59 @@ func (s *Raw) Reverse( /*@ ghost ubuf []byte @*/ ) (p path.Path, err error) {
 }
 
 // ToDecoded transforms a scion.Raw to a scion.Decoded.
-// @ preserves s.Mem(ubuf)
-// @ preserves slices.AbsSlice_Bytes(ubuf, 0, len(ubuf))
+// @ requires  slices.AbsSlice_Bytes(ubuf, 0, len(ubuf))
+// @ requires  s.Mem(ubuf)
+// @ ensures   slices.AbsSlice_Bytes(ubuf, 0, len(ubuf))
+// @ ensures   s.Mem(ubuf)
```

Comment:
why are we removing the `preserves` clauses here, in favor of an equivalent idiom that's twice as big?